### ISSUE TITLE
Directional Disturbance Configuration

### ISSUE THREAD

**Issue by jklaiber** [wemulate/wemulate#53]
**Is your feature request related to a problem? Please describe.**
Yes, the different disturbtion should be configurable in a directed manor. In means of e.g. The traffic vom Interface LAN-A to interface LAN-B should be influenced with a delay, but the from LAN-B to LAN-A there should be no disturbtion. 

**Describe the solution you'd like**
We have to add new configuration parameters to the cli tool e.g.
```
$ wemulate add parameter -n test -d 100 backward -j 10 forward
$ wemulate add parameter -n test -d 100 b -j 10 f
```
The output of the show commands should be configured accordingly e.g.
```
+---------+----------------+----------------+--------------------+
| NAME    | 1. INTERFACE   | 2. INTERFACE   | PARAMETERS         |
+=========+================+================+====================+
| test    | LAN-A          | LAN-B          | --> jitter: 100    |
|         |                |                | <-- delay: 100     |
|         |                |                | <-> bandwidth: 100 |
+---------+----------------+----------------+--------------------+
| test2   | LAN-C          | LAN-D          |                    |
+---------+----------------+----------------+--------------------+
```
The `set` command will be changed accordingly.  
The `bidirectional` value in the `ConnectionModel` can be removed, because the direction will be given directly to the parameters. 
We have to add a new `direction` value to the `ParameterModel`.

**Observation with tcset**
When the parameters are applied without any direction then the traffic will be influenced on the way back to the source. `tcset` will apply the configuration per default with the `--outgoing` argument. 

```
                         +----------------------------------------------+
                         |                                              |
                         |                                              |
                         |                                              |
                         |                                              |
                         |                                              |
                         |                                              |
                         |                                              |
                         |                                              |
                         |    +-----------+           +-----------+     |
                         |    |           |           |           |     |
                         |    |   LAN-A   |           |   LAN-B   |     |
                         |    |           |           |           |     |
                         |    |           |           |           |     |
                         |    |      +----+           |           |     |
                         +----+---^--++++++-----------+--^----+---+-----+
                                  |  +-+-+               |    |
                                  |    | --outgoing      |    |
                                  |    | will influence  |    |
                                  |    | traffic here    |    |
                      request     |    |                 |    |
+----------------+----------------+    |                 |    |            +---------------+
|                |                     |                 |    |  request   |               |
|  Source        |    reply            |                 |    +------------>  Destination  |
|                <---------------------+                 |                 |               |
|                |                                       |      reply      |               |
|                |                                       +-----------------+               |
|                |                                                         |               |
|                |                                                         |               |
|                |                                                         |               |
|                |                                                         |               |
+----------------+                                                         +---------------+
```
```
                         +----------------------------------------------+
                         |                                              |
                         |                                              |
                         |                                              |
                         |                                              |
                         |                                              |
                         |                                              |
                         |                                              |
                         |                                              |
                         |    +-----------+           +-----------+     |
                         |    |           |           |           |     |
                         |    |   LAN-A   |           |   LAN-B   |     |
                         |    |           |           |           |     |
                         |    |           |           |           |     |
                         |    | +---+     |           |           |     |
                         +----+-++^++--+--+-----------+--^----+---+-----+
                     --incoming +-+-+  |                 |    |
                     influences   |    |                 |    |
                     traffic here |    |                 |    |
                                  |    |                 |    |
                      request     |    |                 |    |
+----------------+----------------+    |                 |    |            +---------------+
|                |                     |                 |    |  request   |               |
|  Source        |    reply            |                 |    +------------>  Destination  |
|                <---------------------+                 |                 |               |
|                |                                       |       reply     |               |
|                |                                       +-----------------+               |
|                |                                                         |               |
|                |                                                         |               |
|                |                                                         |               |
|                |                                                         |               |
+----------------+                                                         +---------------+
```

**Comment by jklaiber** [comment]
We are using a new format in order to define the different connections: 

**add**:  
`wemulate add parameter -n test -d 100 -j 100 -src LAN-A -dst LAN-B`. 
`wemulate add parameter -n test -d 200 -j 10 -src LAN-B -dst LAN-A`. 
  
**set**:  
The same goes for the set command. The set command will only overwrite the specific direction. Only when there is no connection given, than the set command will overwrite the whole connection.

**Comment by severindellsperger** [comment]
Similar to the above-mentioned commands we will implement the other ones:

## delete
``` wemulate delete parameter -n test -j -src LAN-A -dst LAN-B ``` will delete the jitter parameter in direction LAN-A to LAN-B.
``` wemulate delete parameter -n test -l ``` will delete the packet loss parameter in both directions (if there is any)

## reset
The reset command will still delete all parameters. In the new version, it will delete all parameters in both directions.

**Comment by severindellsperger** [comment]
## restore
We also introduce a `restore` command, which allows restoring the config, which can be found in the database.
To do so we use the ConfigController and the command 
- restore all connections and parameter (the whole device): ``` wemulate restore device```
- restore a specific connection and its parameters ```wemulate restore connection -n connection_name```

**Comment by severindellsperger** [comment]
This issue was closed by the release of version 2.0.0
